Automattic/wp-calypso
# Happy Chat: Show image files in timeline

#### Changes proposed in this Pull Request

This PR mirrors changes that have been made in the `happychat-client` repo here: https://github.com/Automattic/happychat-client/pull/337

Adds file viewing to the Happy Chat client. This does not allow customers to upload files.

- The Timeline component passes files from messages into a new `MessageFiles` component, which displays `ImageFile`s at the bottom of the message
- `ImageFile` makes an authenticated request for the file binary data, creates a URL for the blob, and shows that in an `<img>`. Using Blobs lets this work even when third-party cookies are disabled.
- Clicking the image also opens the blob URL in a new tab — so the user does not need to re-auth anywhere to see the full-size image.

Importantly, we're assuming for now that all files are images, which is true for the time being. But if we added other file types, it shouldn't be too hard for MessageFiles to check a file type and use different components to render each.

#### Testing instructions

Make sure your local Calypso is connecting to Happy Chat's staging server, then:

- Open staging HUD and go green
- Start a chat from local Calypso
- From the HUD upload some images and send them in a message
- From the customer client — check that the messages render well, you can click on them, etc

Here's a video example:

https://user-images.githubusercontent.com/518059/137175130-387faf3e-535d-4472-bc05-cf97ca10bc1e.mov
